lib/ops/opsservice/shrink.go

```diff
@@ -406,6 +410,23 @@ func (s *site) shrinkOperationStart(ctx *operationContext) (err error) {
 	return nil
 }
 
+func (s *site) pickShrinkMasterRunner(ctx *operationContext, removedServer storage.Server) (*serverRunner, error) {
+	masters, err := s.getTeleportServers(schema.ServiceLabelRole, string(schema.ServiceRoleMaster))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	// Pick any master server except the one that's being removed.
+	for _, master := range masters {
+		if master.IP != removedServer.AdvertiseIP {
+			return &serverRunner{
+				&master, &teleportRunner{ctx, s.domainName, s.teleport()},
+			}, nil
+		}
+	}
+	return nil, trace.NotFound("could not find appropriate master server among %v",
```

Comment:
The message could be more specific:
```
%v is being removed and no more master nodes are available to execute the operation
```
which is kind of weird since there's already a check elsewhere that verifies that the last master node cannot be shrunk - it only means teleport can be flaky and not return all master nodes when queried (?)